operations/kubernetes/csm-service-upgrade-FAILED.md

```diff
@@ -0,0 +1,193 @@
+    response = http.request(method=method, url=url, headers=headers, **kwargs)
+  File "/usr/lib/python3.8/site-packages/requests/sessions.py", line 542, in request
+    resp = self.send(prep, **send_kwargs)
+  File "/usr/lib/python3.8/site-packages/requests/sessions.py", line 655, in send
+    r = adapter.send(request, **kwargs)
+  File "/usr/lib/python3.8/site-packages/requests/adapters.py", line 507, in send
+    raise RetryError(e, request=request)
+requests.exceptions.RetryError: HTTPConnectionPool(host='cray-dhcp-kea-api', port=8000): Max retries exceeded with url: / (Caused by ResponseError('too many 503 error responses'))
+```
+Looking for the Kea we noticed that is also in bad state, it is reporting "failed to select a subnet...." messages meaning it hasn't loaded the config.
+
+```sh
+ncn-m001:~ # kubectl -n services logs cray-dhcp-kea-84b4dd8c5d-ng4mc -c cray-dhcp-kea | head
+2022-04-19 13:29:25.948 ERROR [kea-dhcp4.bad-packets/14.139702139406208] DHCP4_PACKET_NAK_0001 [hwtype=1 00:40:a6:83:51:a6], cid=[ff:a6:83:51:a6:00:01:00:01:29:ea:f0:a5:00:40:a6:83:51:a6], tid=0x4cef7c4: failed to select a subnet for incoming packet, src 10.2.0.5, type DHCPDISCOVER
+2022-04-19 13:29:25.999 ERROR [kea-dhcp4.bad-packets/14.139702139406208] DHCP4_PACKET_NAK_0001 [hwtype=1 00:40:a6:83:51:48], cid=[ff:a6:83:51:48:00:01:00:01:29:ea:f0:76:00:40:a6:83:51:48], tid=0x4ec73527: failed to select a subnet for incoming packet, src 10.2.0.5, type DHCPDISCOVER
+2022-04-19 13:29:26.023 ERROR [kea-dhcp4.bad-packets/14.139702139406208] DHCP4_PACKET_NAK_0001 [hwtype=1 00:40:a6:83:02:8b], cid=[ff:a6:83:02:8b:00:01:00:01:29:ea:f0:65:00:40:a6:83:02:8b], tid=0x18dba25f: failed to select a subnet for incoming packet, src 10.2.0.5, type DHCPDISCOVER
+```
+
+
+Kea is getting a 503 when trying to communicate with SLS so it can't retrieve the list of networks to build the configuration
+```sh
+/ $ /srv/kea/dhcp-helper.py
+ERROR: 503 Server Error: Service Unavailable for url: http://cray-sls/v1/search/hardware?type=comptype_cabinet
+```
+
+Checking SLS and its running fine.
+
+```sh
+ncn-m001:~ # cray sls dumpstate list --format json | head
+{
+  "Hardware": {
+    "d0w1": {
+      "Parent": "d0",
+      "Xname": "d0w1",
+      "Type": "comptype_cdu_mgmt_switch",
+      "Class": "Mountain",
+      "TypeString": "CDUMgmtSwitch",
+      "LastUpdated": 1618431190,
+      "LastUpdatedTime": "2021-04-14 20:13:10.082771 +0000 +0000",
+      
+```
+It can't be queried from the Kea pod
+
+```sh
+$ curl http://cray-sls/v1/search/hardware?type=comptype_cabinet
+upstream connect error or disconnect/reset before headers. reset reason: connection failure
+```
+
+The same query from Unbound works OK
+
+```sh
+$ curl http://cray-sls/v1/search/hardware?type=comptype_cabinet
+[{"Parent":"s0","Children":["x3000c0r42b0","x3000c0r41b0","x3000c0r39b0","x3000m0","x3000c0r40b0"] 
+```
+
+Looks like the istio-proxy container of the cray-dhcp-kea pod isn't behaving correctly
```

Comment:
7. Check to see if the `istio-proxy` container of the `cray-dhcp-kea` pod is not functioning.

    ```sh
    ncn-m001# kubectl -n services logs cray-dhcp-kea-84b4dd8c5d-ng4mc -c istio-proxy --tail=10
    [2022-04-19T13:35:23.864Z] "POST / HTTP/1.1" 503 UF,URX "-" "TLS error: Secret is not supplied by SDS" 52 91 18 - "-" "curl/7.79.1" "622c1d7d-b9bf-4fea-9bc2-28dfe57ae9f0" "cray-dhcp-kea-api:8000" "10.37.0.73:8000" outbound|8000||cray-dhcp-kea-api.services.svc.cluster.local - 10.27.72.242:8000 10.37.0.73:58716 - default
    [2022-04-19T13:35:25.005Z] "GET /metrics HTTP/1.1" 503 UF "-" "-" 0 91 0 - "-" "Prometheus/2.18.1" "272cb1cb-483d-4c18-bb25-924a7023ce15" "10.37.0.73:8080" "127.0.0.1:8080" inbound|8080|exporter|cray-sysmgmt-health-dhcp-kea-exporter.services.svc.cluster.local - 10.37.0.73:8080 10.43.0.71:49934 - default
    [2022-04-19T13:35:55.308Z] "GET /metrics HTTP/1.1" 503 UF "-" "-" 0 91 0 - "-" "Prometheus/2.18.1" "9a73fa92-2357-4b67-9901-4920ec54c72a" "10.37.0.73:8080" "127.0.0.1:8080" inbound|8080|exporter|cray-sysmgmt-health-dhcp-kea-exporter.services.svc.cluster.local - 10.37.0.73:8080 10.43.0.71:49934 - default
    [2022-04-19T13:36:07.206Z] "POST / HTTP/1.1" 503 UF,URX "-" "TLS error: Secret is not supplied by SDS" 52 91 42 - "-" "curl/7.79.1" "711361e4-13a1-4457-b80b-3315ee42d460" "cray-dhcp-kea-api:8000" "10.37.0.73:8000" outbound|8000||cray-dhcp-kea-api.services.svc.cluster.local - 10.27.72.242:8000 10.37.0.73:60928 - default
    [2022-04-19T13:36:18.716Z] "POST / HTTP/1.1" 503 UF,URX "-" "TLS error: Secret is not supplied by SDS" 52 91 45 - "-" "curl/7.79.1" "120f0e1a-4aec-4135-b466-cd87b5bf41e6" "cray-dhcp-kea-api:8000" "10.37.0.73:8000" outbound|8000||cray-dhcp-kea-api.services.svc.cluster.local - 10.27.72.242:8000 10.37.0.73:33464 - default
    [2022-04-19T13:36:23.852Z] "POST / HTTP/1.1" 503 UF,URX "-" "TLS error: Secret is not supplied by SDS" 52 91 53 - "-" "curl/7.79.1" "e2626e00-6537-4e55-9830-fcaf10805cf1" "cray-dhcp-kea-api:8000" "10.37.0.73:8000" outbound|8000||cray-dhcp-kea-api.services.svc.cluster.local - 10.27.72.242:8000 10.37.0.73:33558 - default
    [2022-04-19T13:36:25.005Z] "GET /metrics HTTP/1.1" 503 UF "-" "-" 0 91 0 - "-" "Prometheus/2.18.1" "cd8d304f-714b-4178-b1d1-69720fb7b5c3" "10.37.0.73:8080" "127.0.0.1:8080" inbound|8080|exporter|cray-sysmgmt-health-dhcp-kea-exporter.services.svc.cluster.local - 10.37.0.73:8080 10.43.0.71:49934 - default
    [2022-04-19T13:36:55.005Z] "GET /metrics HTTP/1.1" 503 UF "-" "-" 0 91 0 - "-" "Prometheus/2.18.1" "040b2e40-896b-49f9-9b2a-06861c58de8a" "10.37.0.73:8080" "127.0.0.1:8080" inbound|8080|exporter|cray-sysmgmt-health-dhcp-kea-exporter.services.svc.cluster.local - 10.37.0.73:8080 10.43.0.71:49934 - default
    [2022-04-19T13:37:07.186Z] "POST / HTTP/1.1" 503 UF,URX "-" "TLS error: Secret is not supplied by SDS" 52 91 26 - "-" "curl/7.79.1" "6407b6fd-3b1b-4d4b-9685-23b42b962a08" "cray-dhcp-kea-api:8000" "10.37.0.73:8000" outbound|8000||cray-dhcp-kea-api.services.svc.cluster.local - 10.27.72.242:8000 10.37.0.73:35484 - default
    [2022-04-19T13:37:18.718Z] "POST / HTTP/1.1" 503 UF,URX "-" "TLS error: Secret is not supplied by SDS" 52 91 18 - "-" "curl/7.79.1" "5ee53ea6-a8fc-4cb6-9887-c196626f74d8" "cray-dhcp-kea-api:8000" "10.37.0.73:8000" outbound|8000||cray-dhcp-kea-api.services.svc.cluster.local - 10.27.72.242:8000 10.37.0.73:36314 - default
    ```